src/parser/tests/errors.rs

```diff
@@ -53,7 +55,7 @@ fn wrong_end_block() {
         "{{ hey %}",
         &[
             "1:9",
-            "expected an integer, a float, a string, `true` or `false`, an identifier, a dotted identifier (identifiers separated by `.`), or an expression"
+            "expected an integer, a float, a string, `true` or `false`, an identifier, a dotted identifier (identifiers separated by `.`), an identifier (identifiers separated by `.` or `[]`s), or an expression"
```

Comment:
the confusing message I was referring above: `an identifier, ... , an identifier (...),`

Reply:
I agree that this is a confusing message. But I'm not sure how best to fix it. As an error it is listing everything that it could expect to see unfortunately this means it lists identifier 3 times. 

I'll try to resolve this with the next commit - please let me know your thoughts on it